### PR TITLE
fromString expects context and tenantTable to be URL encoded; toString URL encodes

### PR DESCRIPTION
**Pull request by horwitzj**

### CODE REVIEW

File: src/main/java/com/salesforce/dynamodbv2/mt/util/StreamArn.java [hunk 1]
```diff
@@ -18,28 +23,43 @@
...
-        MtStreamArn(String prefix, String tableName, String streamLabel, String context, String tenantTableName) {
+        MtStreamArn(String prefix,
+            String tableName,
+            String streamLabel,
+            String slashFreeContext,
+            String slashFreeTenantTableName) {
             super(prefix, tableName, streamLabel);
-            this.context = context;
-            this.tenantTableName = tenantTableName;
+            Preconditions.checkArgument(!slashFreeContext.contains("/"),
```
Comment (busjaeger) [inline src/main/java/com/salesforce/dynamodbv2/mt/util/StreamArn.java]: I was expecting the MtStreamArn to store the un-encoded values in memory and only encode/decode when going to/from string.

File: src/test/java/com/salesforce/dynamodbv2/mt/util/StreamArnTest.java [hunk 2]
```diff
@@ -1,29 +1,40 @@
...
-        String virtualTableName = "books_table";
+        String context = "account/tenant1";
+        String virtualTableName = "books_table/x";
+
+        final String escapedContext = URLEncoder.encode(context, StandardCharsets.UTF_8.name());
+        final String escapedVirtualTableName = URLEncoder.encode(virtualTableName, StandardCharsets.UTF_8.name());
 
         String expectedString1 =
             "arn:" + qualifier + "table/" + tableName + "/stream/"
                 + streamLabel;
```
[the remark this answers is not in the document]
Comment (horwitzj) [reply src/test/java/com/salesforce/dynamodbv2/mt/util/StreamArnTest.java]: old code—not sure

File: src/main/java/com/salesforce/dynamodbv2/mt/util/StreamArn.java [hunk 3]
```diff
@@ -62,13 +82,33 @@ public int hashCode() {
...
         @Override
         public String toString() {
             return String.format(VIRTUAL_FORMAT, super.toString(), context, tenantTableName);
         }
 
     }
 
+    // TODO: don't decode every time (i.e., do it once and save it)?
+    // unchecks potential UnsupportedEncodingException (as IllegalArgumentException)
+    private static String wrappedDecoder(String s) {
```
[the remark this answers is not in the document]
Comment (horwitzj) [reply src/main/java/com/salesforce/dynamodbv2/mt/util/StreamArn.java]: it also unchecks the (potential) checked (UnsupportedEncodingException) exception (and if we're not supporting UTF8, we probably have bigger problems...)

[the remark this answers is not in the document]
Comment (busjaeger) [reply src/main/java/com/salesforce/dynamodbv2/mt/util/StreamArn.java]: Maybe 'encodeUnchecked' would be a better name?

**Pull request merged**